apps/transport/lib/config/runtime_configuration.ex

```diff
@@ -0,0 +1,54 @@
+defmodule Transport.RuntimeConfiguration do
+  @moduledoc """
+  A module allowing us to bring some structure & testing to what is happening
+  inside `runtime.exs`.
+  """
+
+  defmodule SystemEnvProvider do
+    @moduledoc """
+    A behaviour wrapping the sub-part of `System.get_env`  calls required by
+    the current `runtime.exs` configuration. We define this boundary/interface
+    to help stub the environment variables during tests, and ensure we can get
+    robust well-tested production configuration.
+    """
+
+    @callback get_env(String.t()) :: String.t() | nil
+    @callback get_env(String.t(), String.t()) :: String.t() | nil
```

Comment:
```suggestion
    @callback get_env(String.t()) :: String.t()
    @callback get_env(String.t(), String.t()) :: String.t()
```

Pour correspondre à [la librairie standard](https://hexdocs.pm/elixir/System.html#get_env/2). On ne peut pas avoir `nil` sauf si on passe cette valeur en fall back

Reply:
De ce que je lis de la doc standard, le get_env/1 renvoie un `String.t() | nil`

Reply:
Seulement si on passe `nil` en valeur par défaut

```
@spec get_env(String.t(), String.t()) :: String.t()
@spec get_env(String.t(), nil) :: String.t() | nil
```

Reply:
Oui justement, mon commentaire portait sur la première ligne de ta suggestion de diff

Reply:
Si je reprends [la spec](https://hexdocs.pm/elixir/System.html#get_env/2), on a comme écrit un peu plus haut:

```elixir
@spec get_env(String.t(), String.t()) :: String.t()
@spec get_env(String.t(), nil) :: String.t() | nil
```

Je comprends ce que veut dire @ptitfred, la suggestion de changement proposait quelque chose de différent:

```elixir
@callback get_env(String.t()) :: String.t()
@callback get_env(String.t(), String.t()) :: String.t()
```

Je m'aligne sur la spec du coup.

Je me demande ce que ça va donner avec Elixir 1.17 ce genre de choses d'ailleurs, en aparté. Les specs actuelles n'expriment pas qu'on peut laisser de côté le deuxième argument.

Reply:
Je n'avais fait que suivre les signatures pour arriver à cette conclusion. Un vrai typechecker serait précieux et plus précis.